xla/stream_executor/command_buffer.h

```diff
@@ -47,108 +48,52 @@ class Stream;
-  // the `kDefaultExecutionScope` execution scope.
-  //
-  // Example #1: independent execution scopes and independent barriers
-  //
-  // ExecutionScope #0       ExecutionScope #1
-  //
-  //          A                        D
-  //          B                        E
-  // ----- barrier -----      ----- barrier -----
-  //          C                        F
-  //
-  //   (1) Commands A and B can run concurrently and must complete before C.
-  //   (2) Commands D and E can run concurrently and must complete before F.
-  //   (3) There is no syncrhonization between execution scopes, and commands
-  //       from different execution scopes can execute concurrently with each
-  //       other as long as they satisfy constraints of their respective
-  //       execution scopes.
-  //
-  //
-  //
-  // Example #2: dependencies between scopes and inter-scope barriers
-  //
-  // ExecutionScope #0       ExecutionScope #1
-  //
-  //          A                        D
-  //          B                        E
-  // ----------------- barrier ------------------
-  //          C                        F
-  //
-  //   (1) Commands A and B can run concurrently and must complete before
-  //       C and F.
-  //   (2) Commands D and E can run concurrently and must complete before
-  //       C and F.
-  //   (3) Commands C and F can run concurrently.
-  //   (4) All commands before a shared barrier (in both excecution scopes)
-  //       should complete before any command after a berrier starts execution.
-  //
-  //
-  //
-  // Example #3: one-directional barriers between execution scopes
-  //
-  // ExecutionScope #0       ExecutionScope #1
-  //
-  //          A
-  //          B
-  // ----- barrier -----               D
-  //          C            \           E
-  //                           ----- barrier -----
-  //                                   F
-  //
-  //   (1) Commands A and B can run concurrently and must complete before
-  //       C and F.
-  //   (2) Commands D and E can run concurrently and must complete before
-  //       F (does not synchronize with C).
-  //   (3) Commands C and F can run concurrently.
-  //
-  //  This is a more fine-grained barrier than in example #2: it enforces
-  //  synchronization from execution scope #0 to execution scope #1 but no
-  //  synchronization in other direction. For CUDA/ROCM backend it has the same
-  //  semantics as stream wait operation.
-  //
-  TSL_LIB_GTL_DEFINE_INT_TYPE(ExecutionScopeId, uint64_t);
-  static constexpr auto kDefaultExecutionScope = ExecutionScopeId(0);
+  // std::vector<CommandBufferNodeHandle> are created by node factory functions
+  // and can be referenced in node update functions. The handle has the same
+  // properties as a pointer (can be constructed from a nullptr, trivial
+  // copyable, POD, etc.), that's why we use a pointer to define it.
+
+  // CommandBufferNodeHandleOpaque is an opaque type that won't be ODR used,
```

Comment:
For types and aliases under `CommandBuffer` no need to prefix them with `CommandBufferXYZ`, i.e. just `NodeHandleOpaque` is good enough

Reply:
Will make the fix